clientconn.go

```diff
@@ -170,17 +170,17 @@ func WithInsecure() DialOption {
 
 // WithTransportCredentials returns a DialOption which configures a
 // connection level security credentials (e.g., TLS/SSL).
-func WithTransportCredentials(creds credentials.TransportAuthenticator) DialOption {
+func WithTransportCredentials(auth credentials.TransportAuthenticator) DialOption {
```

Comment:
it looks a bit weird to me have a function called `withTransportCredentials` that takes a `TransportAuthenticator` object.
In C/C++ we call these `ChannelCredentials` so maybe we could align here with:
```
func WithChannelCredentials(auth credentials.ChannelCredentials) DialOption
```
or just be consistent
```
func WithTransportAuthenticator(auth credentials.TransportAuthenticator) DialOption
```